sycl/test-e2e/FreeFunctionKernels/get_kernel_info.cpp

```diff
@@ -0,0 +1,120 @@
+// REQUIRES: aspect-usm_shared_allocations
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// UNSUPPORTED: cuda, hip
+// UNSUPPORTED-INTENDED: Not implemented yet for Nvidia/AMD backends.
+
+#include <iostream>
+#include <sycl/ext/oneapi/free_function_queries.hpp>
+#include <sycl/ext/oneapi/get_kernel_info.hpp>
+#include <sycl/kernel_bundle.hpp>
+#include <sycl/usm.hpp>
+
+namespace syclext = sycl::ext::oneapi;
+namespace syclexp = sycl::ext::oneapi::experimental;
+
+static constexpr size_t NUM = 1024;
+static constexpr size_t WGSIZE = 16;
+
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((syclexp::work_group_size<WGSIZE>))
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((syclexp::nd_range_kernel<1>))
+void func(float start, float *ptr) {
+  size_t id = syclext::this_work_item::get_nd_item<1>().get_global_linear_id();
+  ptr[id] = start + static_cast<float>(id);
+}
+
+bool check_result(int *ptr) {
+  for (size_t i = 0; i < NUM; ++i) {
+    const int expected = 3 + static_cast<int>(i);
+    if (ptr[i] != expected)
+      return true;
+  }
+  return false;
+}
+
+static bool call_kernel_code(sycl::queue &q, sycl::kernel &kernel) {
+  int *ptr = sycl::malloc_shared<int>(NUM, q);
+  q.submit([&](sycl::handler &cgh) {
+     cgh.set_args(3, ptr);
+     sycl::nd_range ndr{{NUM}, {WGSIZE}};
+     cgh.parallel_for(ndr, kernel);
+   }).wait();
+  const bool ret = check_result(ptr);
+  sycl::free(ptr, q);
+  return ret;
+}
+
+bool test_ctxt_dev(sycl::kernel &k, sycl::queue &q) {
+  const auto wg_size_cmp =
+      k.get_info<sycl::info::kernel_device_specific::work_group_size>(
+          q.get_device());
+  const auto wg_size = syclexp::get_kernel_info<
+      func, sycl::info::kernel_device_specific::work_group_size>(
+      q.get_context(), q.get_device());
+  if (wg_size_cmp != wg_size)
+    std::cerr << "Work group size from get_info: " << wg_size_cmp
+              << " is not equal to work group size from get_kernel_info: "
+              << wg_size << std::endl;
+  return wg_size_cmp == wg_size;
+}
+
+bool test_ctxt(sycl::kernel &k, sycl::queue &q) {
```

Comment:
Shouldn't this function check that k.get_info matches the result from the new queries like the others do?